spec/unit/radius_ip_whitelist_spec.rb

```diff
@@ -0,0 +1,88 @@
+describe RadiusIpWhitelist do
+  let(:organisation) { create(:organisation) }
+  let(:location1) { create(:location, organisation: organisation) }
+  let(:location2) { create(:location, organisation: organisation) }
+
+  before do
+    create(:ip, address: '1.1.1.1', location: location1)
+    create(:ip, address: '1.2.2.1', location: location1)
+    create(:ip, address: '2.2.2.2', location: location2)
+
+    location1.update(radius_secret_key: 'radkey1')
+    location2.update(radius_secret_key: 'radkey2')
+  end
+
+  let(:configuration_result) { subject.execute }
+
+  describe '#execute' do
+    it 'returns the correct configuration' do
+      expect(configuration_result).to eq('client 1-1-1-1 {
+  ipaddr = 1.1.1.1
+  secret = radkey1
+}
+client 1-2-2-1 {
```

Comment:
Thought about whether we should pull out these `client` blocks into some structure to reduce duplication in this suite, and maybe making the fact the spacing is necessary clearer.

Don't think this test file is large enough yet for it to be a problem though.